packages/deployments/contracts/src/config/configureProtocol.ts

```diff
@@ -0,0 +1,122 @@
+// NOTE: Agents will currently be whitelisted on ALL domains.
+type WhitelistAgents = {
+  // Arrays of addresses for each type of agent that requires whitelisting.
+  relayers: string[]; // NOTE: Relayers will be whitelisted for both `execute` and messaging calls.
+  sequencers: string[];
+  routers: string[];
+};
+
+type AssetStack = {
+  canonical: {
+    // The canonical domain of the asset.
+    domain: string;
+    // Address of the bridged asset on the canonical domain.
+    local: string;
+    // The address of the official canonical token.
+    // NOTE: If not specified here, then I guess the asset is natively cross-chain?
+    adopted?: string;
+  };
+  representational: {
+    [domain: string]: {
+      // Address of the bridged asset on this domain.
+      local: string;
+      // Address of the adopted asset on this domain.
+      // NOTE: If adopted is specified, a stableswap will be initialized! If not
+      // specified, then we assume the local asset is the adopted asset on this domain.
+      adopted?: string;
+    };
+  };
+};
+
+type DomainStack = {
+  // Meta info.
+  chain: number;
+  domain: string;
+
+  // RPC provider to use for this network.
+  rpc: string;
+
+  deployments: {
+    // Diamond.
+    Connext: string;
+    // Handlers.
+    BridgeRouter: string;
+    RelayerFeeRouter: string;
+    PromiseRouter: string;
+    // Registry.
+    TokenRegistry: string;
+
+    // Messaging Layer.
+    // ConnectorManager
+    // SendOutboundRootResolver
+
+    // The messaging layer deployments are different depending on whether this
+    // is the hub domain or spoke domain.
+    messaging:
+      | {
+          RootManager: string;
+          MainnetConnector: string;
+          HubConnectors: string;
+        }
+      | {
+          SpokeConnector: string;
+        };
+  };
+};
+
+type ProtocolStack = {
+  hub: string; // The hub domain.
+  // Domain stack should have all info pertaining to each supported domain.
+  domains: DomainStack[];
+  // Crosschain ERC20 assets to enroll in TokenRegistry.
+  assets: AssetStack[];
+  // Agents that need to be whitelisted (across all domains).
+  // Leave undefined if no agents should be whitelisted in this setup.
+  agents?: WhitelistAgents;
+};
+
+/**
+ * Handle configuration of the entire protocol, including messaging stack and connext diamond
+ * contracts, across all listed domains.
+ *
+ * Should effectively be a diagnostic on the whole protocol, making sure that everything that
+ * requires configuration and/or setup has been done so properly.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const configureProtocol = async (protocol: ProtocolStack) => {
```

Comment:
I don't think this should be in `src/config` just the name is a bit confusing since config means like configuration, but this is more like an initialization.

I like the idea of this, and I think this is something that should be able to be run by both hardhat tasks and integration tests so that we can run it on prod and local envs.

Reply:
sure, I think init is a better name as well

We can work towards using both the `protocol` argument and a JSON config file on what to use here. Obviously I could just be grabbing deployments directly... But I'll leave that to the consumer (hardhat task or integration test, etc)